include/aws/http/proxy_strategy.h

```diff
@@ -228,23 +212,27 @@ struct aws_http_proxy_strategy_factory_tunneling_kerberos_options {
 
 struct aws_http_proxy_strategy_factory_tunneling_adaptive_kerberos_options {
     struct aws_http_proxy_strategy_factory_tunneling_kerberos_options kerberos_options;
-    
+ };
 
-};
+/*SA-Added Start*/
+ struct aws_http_proxy_strategy_factory_kerberos_auth_config {
+
+     /* type of proxy connection being established, must be forwarding or tunnel */
+     enum aws_http_proxy_connection_type proxy_connection_type;
+
+     /* user token to use in kerberos authentication which is base64 encoded and provided by user*/
+     struct aws_byte_cursor user_token;
```

Comment:
I think it would be better to pass in an (async-ready) function that fetches the user token rather than the user token itself.  As it stands, I don't think this would support long-term connections because eventually the token expires and there's no sensible way to get a refreshed one.